### PR TITLE
Fix flaky failure related to core_device_list.json not being found

### LINKED ISSUE #141892: [flutter_tools] PathNotFoundException: Cannot open file, path = '.../core_device_list.json' (OS Error: No such file or directory, errno = 2)
This is happening both in the wild and on CI.

A crash report in beta/3.19.0-0.2.pre:

```
PathNotFoundException: Cannot open file, path = '/var/folders/kl/hdgxgrc93gs870zyrkz0dc_m0000gn/T/flutter_tools.iWDQmn/core_devices.RElIQa/core_device_list.json' (OS Error: No such file or directory, errno = 2)
at _File.throwIfError(file_impl.dart:675)
at _File.openSync(file_impl.dart:490)
at _File.readAsBytesSync(file_impl.dart:574)
at _File.readAsStringSync(file_impl.dart:624)
at ForwardingFile.readAsStringSync(forwarding_file.dart:99)
at IOSCoreDeviceControl._listCoreDevices(core_devices.dart:83)
at <asynchronous gap>(async)
at IOSCoreDeviceControl.getCoreDevices(core_devices.dart:114)
at <asynchronous gap>(async)
at XCDevice.getAvailableIOSDevices(xcdevice.dart:487)
at <asynchronous gap>(async)
at IOSDevices.startPolling(devices.dart:108)
at <asynchronous gap>(async)
```

The CI overall_experience_test.dart flake:

```
 10:21 +54 -1: test/integration.shard/overall_experience_test.dart: flutter run handle SIGUSR1/2 [E]                                                                                                    
   Expected: [
               'Performing hot reload...                                        ',
               <a string starting with 'Reloaded 0 libraries in '>,
               'Performing hot restart...                                       ',
               <a string starting with 'Restarted application in '>,
               '',
               'Application finished.',
               'ready'
             ]
     Actual: WhereIterable<String>:[
               'Performing hot reload...                                        ',
               'Reloaded 0 libraries in 45ms (compile: 4 ms, reload: 0 ms, reassemble: 31 ms).',
               'Performing hot restart...                                       ',
               'Restarted application in 247ms.',
               '',
               'Application finished.',
               '',
               'Oops; flutter has exited unexpectedly: "PathNotFoundException: Cannot open file, path = \'/Volumes/Work/s/w/ir/x/t/flutter_tools.OnrQMu/core_devices.VRAibL/core_device_list.json\' (OS Error: No such file or directory, errno = 2)".',
               'ready'
             ]
      Which: at location [6] is '' instead of 'ready'
   
   package:matcher                                            expect
   test/integration.shard/overall_experience_test.dart 111:7  main.<fn>
```
Seen in presubmit 
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8758444597475340193/+/u/run_test.dart_for_tool_integration_tests_shard_and_subshard_4_4/test_stdout

cc @vashworth

### PR DESCRIPTION
**Pull request by bkonyi**
It's possible that the tool can be in the process of shutting down, which could result in the temp directory being deleted after the shutdown hooks run before we check if `output` exists. If this happens, we shouldn't crash but just carry on as if no devices were found as the tool will exit on its own.

Fixes https://github.com/flutter/flutter/issues/141892

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/core_devices.dart [hunk 1]
```diff
@@ -79,14 +79,24 @@ class IOSCoreDeviceControl {
...
+      final bool isPossiblyShutdown = _fileSystem is LocalFileSystem && _fileSystem.disposed;
+
+      // It's possible that the tool is in the process of shutting down, which
+      // could result in the temp directory being deleted after the shutdown hooks run
+      // before we check if `output` exists. If this happens, we shouldn't crash
+      // but just carry on as if no devices were found as the tool will exit on
+      // its own.
+      //
+      // See https://github.com/flutter/flutter/issues/141892 for details.
+      if (!isPossiblyShutdown && !output.existsSync()) {
```
Comment (andrewkolos) [inline packages/flutter_tools/lib/src/ios/core_devices.dart:91]: Nit: consider renaming `isPossiblyShutdown` to `isToolPossiblyShutdown`

Comment (andrewkolos) [reply packages/flutter_tools/lib/src/ios/core_devices.dart:91]: or `toolPossbilyShutdown`

Comment (bkonyi) [reply packages/flutter_tools/lib/src/ios/core_devices.dart]: Done.

**Review by andrewkolos — APPROVED** [review]
LGTM

**Pull request merged**